packages/salesforcedx-vscode-core/src/commands/forceApexTestRun.ts

```diff
@@ -81,35 +82,59 @@ export class TestsSelector
   }
 }
 
+export class ForceApexTestRunCommandFactory {
+  private _data: ApexTestQuickPickItem;
+  private _getCodeCoverage: boolean;
+  private builder: SfdxCommandBuilder = new SfdxCommandBuilder();
+  private testRunExecutorCommand: Command;
+
+  constructor(data: ApexTestQuickPickItem, getCodeCoverage: boolean) {
+    this._data = data;
+    this._getCodeCoverage = getCodeCoverage;
+  }
+
+  public constructExecutorCommand(): Command {
+    this.builder = this.builder
+      .withDescription(nls.localize('force_apex_test_run_text'))
+      .withArg('force:apex:test:run');
+
+    switch (this._data.type) {
+      case TestType.Suite:
+        this.builder = this.builder
+          .withFlag('--suitenames', `${this._data.label}`);
+        break;
+      case TestType.Class:
+        this.builder = this.builder
+          .withFlag('--classnames', `${this._data.label}`)
+          .withArg('--synchronous');
+        break;
+      default:
+        break;
+    }
+
+    if (this._getCodeCoverage) {
+      this.builder = this.builder
+        .withArg('--codecoverage');
+    }
+
+    this.builder = this.builder
+      .withFlag('--resultformat', 'human')
+      .withFlag('--loglevel', 'error');
+
+    this.testRunExecutorCommand = this.builder.build();
+    return this.testRunExecutorCommand;
+  }
+
+}
+
 export class ForceApexTestRunExecutor extends SfdxCommandletExecutor<
   ApexTestQuickPickItem
-> {
+  > {
   public build(data: ApexTestQuickPickItem): Command {
-    if (data.type === TestType.Suite) {
-      return new SfdxCommandBuilder()
-        .withDescription(nls.localize('force_apex_test_run_text'))
-        .withArg('force:apex:test:run')
-        .withFlag('--suitenames', `${data.label}`)
-        .withFlag('--resultformat', 'human')
-        .withFlag('--loglevel', 'error')
-        .build();
-    } else if (data.type === TestType.Class) {
-      return new SfdxCommandBuilder()
-        .withDescription(nls.localize('force_apex_test_run_text'))
-        .withArg('force:apex:test:run')
-        .withFlag('--classnames', `${data.label}`)
-        .withFlag('--resultformat', 'human')
-        .withArg('--synchronous')
-        .withFlag('--loglevel', 'error')
-        .build();
-    } else {
-      return new SfdxCommandBuilder()
-        .withDescription(nls.localize('force_apex_test_run_text'))
-        .withArg('force:apex:test:run')
-        .withFlag('--resultformat', 'human')
-        .withFlag('--loglevel', 'error')
-        .build();
-    }
+    const getCodeCoverage: boolean = SfdxCoreSettings.getInstance()
```

Comment:
We should get this directly from the exported sfdxCoreSettings from https://github.com/forcedotcom/salesforcedx-vscode/blob/develop/packages/salesforcedx-vscode-core/src/settings/index.ts#L10 instead of calling `getInstance()`.

Nothing wrong with the way you are doing it here - just a matter of consistency.